app/controllers/questions/personal_info_controller.rb

```diff
@@ -6,16 +6,21 @@ def tracking_data
       {}
     end
 
-    def after_update_success
+    def update
       unless Client.after_consent.where(intake: current_intake).exists?
         routing_service = PartnerRoutingService.new(
           source_param: current_intake.source,
-          zip_code: current_intake.zip_code,
+          zip_code: form_params[:zip_code],
         )
         vita_partner = routing_service.determine_partner
 
-        current_intake.client.update(vita_partner: vita_partner, routing_method: routing_service.routing_method)
-        current_intake.update(vita_partner: vita_partner)
+        if vita_partner.present?
+          current_intake.client.update(vita_partner: vita_partner, routing_method: routing_service.routing_method)
+          current_intake.update(vita_partner: vita_partner)
+          super
+        else
+          return redirect_to at_capacity_questions_path
+        end
```

Comment:
janky? not janky?

Reply:
what are the implications of changing this from after_update_success to update? I'm guessing we did this so that we can do the redirect and that's why we call super, and i actually think that's great and a better pattern than after_update_success, but does not having after_update_success defined mean that it uses the questions_controller implementation of that method and if so does it do anything?

Reply:
It actually feels more in line with Products expectations for us to just toss that data, actually, since they haven't consented to anything, so im kind of cool with you doing that. It might be worth noting that it was an intentional decision as a comment, though.